### PR TITLE
Visual Test Setup exception 

### PR DESCRIPTION
**Pull request by MSR-K**
Added New VisualTestSetupException and Updated Visual.java.

### CODE REVIEW

File: src/main/java/com/znsio/e2e/exceptions/VisualTestSetupException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.znsio.e2e.exceptions;
+
+public class VisualTestSetupException extends RuntimeException{
+    public VisualTestSetupException(String message) {
+        super(message);
+    }
+}
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/exceptions/VisualTestSetupException.java:7]: Add the 2nd constructor as well (similar to other custom exceptions)

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 2]
```diff
@@ -115,10 +116,11 @@ private com.applitools.eyes.appium.Eyes instantiateAppiumEyes(String driverType,
         try {
             appEyes.open(innerDriver, appName + "-" + platform, testName);
             LOGGER.info("instantiateAppiumEyes: Is Applitools Visual Testing enabled? - " + !appEyes.getIsDisabled());
-        } catch(IllegalArgumentException e) {
-            throw new InvalidTestDataException(String.format("Exception in instantiating Applitools for Apps: '%s;", e.getMessage(), e));
+        } catch(IllegalArgumentException | EyesException e) {
+            LOGGER.error("Issue found while instantiating Applitools eye, closing Appium driver instance");
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Visual.java:120]: Create a common string variable for the error message. Use that string variable to log and throw the exception

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 3]
```diff
@@ -92,7 +93,7 @@ private com.applitools.eyes.appium.Eyes instantiateAppiumEyes(String driverType,
         // appiumEyesRunner = new ClassicRunner();
         // appiumEyesRunner.setDontCloseBatches(true);
         com.applitools.eyes.appium.Eyes appEyes = new com.applitools.eyes.appium.Eyes();
-
+        LOGGER.info("Check Point 1");
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Visual.java]: why are these logs required?

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 4]
```diff
@@ -103,22 +104,23 @@ private com.applitools.eyes.appium.Eyes instantiateAppiumEyes(String driverType,
 
         applitoolsLogFileNameForApp = getApplitoolsLogFileNameFor("app");
         appEyes.setLogHandler(new FileLogger(applitoolsLogFileNameForApp, true, isVerboseLoggingEnabled));
-
+        LOGGER.info("Check Point 2");
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Visual.java]: why are these logs required?

File: src/main/java/com/znsio/e2e/exceptions/VisualTestSetupException.java (hunk 1 above)
Comment (MSR-K) [reply src/main/java/com/znsio/e2e/exceptions/VisualTestSetupException.java:7]: ok

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 5]
```diff
@@ -177,8 +179,10 @@ private com.applitools.eyes.selenium.Eyes instantiateWebEyes(String driverType,
         try {
             webEyes.open(innerDriver, appName + "-" + platform, testName, setBrowserViewPortSize);
             LOGGER.info("instantiateWebEyes:  Is Applitools Visual Testing enabled? - " + !webEyes.getIsDisabled());
-        } catch(IllegalArgumentException e) {
-            throw new InvalidTestDataException(String.format("Exception in instantiating Applitools for Web: '%s;", e.getMessage(), e));
+        } catch(IllegalArgumentException | EyesException e) {
+            LOGGER.error("Issue found while instantiating Applitools eye, closing web-driver instance");
```
[the remark this answers is not in the document]
Comment (MSR-K) [reply src/main/java/com/znsio/e2e/tools/Visual.java:182]: ok

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 6]
```diff
@@ -112,13 +113,13 @@ private com.applitools.eyes.appium.Eyes instantiateAppiumEyes(String driverType,
         appEyes.addProperty(TARGET_ENVIRONMENT, String.valueOf(getValueFromConfig(TARGET_ENVIRONMENT)));
         appEyes.addProperty("USER_NAME", USER_NAME);
 
+
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Visual.java:116]: remove unnecessary empty line

File: src/main/java/com/znsio/e2e/tools/Visual.java [hunk 7]
```diff
@@ -112,13 +113,13 @@ private com.applitools.eyes.appium.Eyes instantiateAppiumEyes(String driverType,
...
         try {
             appEyes.open(innerDriver, appName + "-" + platform, testName);
             LOGGER.info("instantiateAppiumEyes: Is Applitools Visual Testing enabled? - " + !appEyes.getIsDisabled());
         } catch(IllegalArgumentException e) {
-            throw new InvalidTestDataException(String.format("Exception in instantiating Applitools for Apps: '%s;", e.getMessage(), e));
+            throw new VisualTestSetupException(String.format("Exception in instantiating Applitools for Apps: '%s;", e.getMessage(), e));
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Visual.java:121]: LOGGER.error missing.

**Pull request merged**